crates/katana/core/src/backend/gas_oracle.rs

```diff
@@ -22,4 +33,89 @@ impl L1GasOracle {
     pub fn current_data_gas_prices(&self) -> GasPrices {
         self.data_gas_prices.clone()
     }
+
+    pub async fn gas_price_worker(
+        &mut self,
+        shutdown_rx: &mut watch::Receiver<()>,
+    ) -> anyhow::Result<()> {
+        let mut buffer = GasPriceBuffer::new();
+        let rpc_url = "https://eth.merkle.io".parse()?;
+        let provider = ProviderBuilder::new().on_http(rpc_url);
+        // every 60 seconds, Starknet samples the base price of gas and data gas on L1
+        let mut interval = tokio::time::interval(INTERVAL); 
+        interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+    
+        loop {
+            tokio::select! {
+                _ = shutdown_rx.changed() => {
+                    break;
+                }
+                // tick every 60 seconds
+                _ = interval.tick() => {
+                    if let Err(e) = update_gas_price(self, provider.clone(), &mut buffer).await {
+                        eprintln!("Error updating gas price: {}", e);
+                    }
+                }
+            }
+        }
+    
+        Ok(())
+    }
```

Comment:
We don' to handle cancellation manually here, the task manager already handles that for us. You can simply just spawn the task on it.

Example:

https://github.com/dojoengine/dojo/blob/082a0e048a0f818a4b61ca1ef3cfcd980695aa1c/crates/katana/node/src/lib.rs#L117-L120

Reply:
Thank you so much for the detailed review @kariy. This is incredibly helpful I’ll start making the suggested changes right away and will let you know when the updates are ready for another look. Thanks again